src/app/modular/post/reducer.ts

```diff
@@ -24,6 +24,9 @@ const reducer: PostReducer = produce(
       case actions.SAVE_POSTS:
         state.posts = action.payload.posts;
         break;
+      case actions.ADD_POST:
+        state.posts[action.payload.post.id] = action.payload.post;
+        break;
```

Comment:
Let's destructure here so the nested objects don't go crazy
```suggestion
      case actions.ADD_POST:
        const { post } = action.payload;
        state.posts[post.id] = post;
        break;
```